test/mocks.py

```diff
@@ -0,0 +1,610 @@
+
+    def mock(*args, **kwargs) -> MagicMock:
+        return create_autospec(*args, **kwargs)
+
+    def get_guild(self) -> MagicMock:
+        '''
+        Get the mock Guild.
+        '''
+
+        return self._guild
+
+    def get_bot(self) -> Bot:
+        '''
+        Get the Bot, for testing bot commands.
+        '''
+
+        return self._bot
+
+    def get_cog(self) -> Cog:
+        '''
+        Get the first Cog on the bot.
+
+        The bot should usually only have one Cog, as added by the @cog() decorator.
+        '''
+
+        cogs = self.get_bot().cogs
+
+        if len(cogs) == 0:
+            raise Exception('Bot has no Cogs')
+
+        return cogs[next(iter(cogs))]
+
+    def get_unique_id(self) -> int:
+        '''
+        Create an ID unique to this run of the application.
+        '''
+
+        global unique_id
+
+        unique_id += 1
+
+        return unique_id
+
+    def find(self, collection, **kwargs) -> Any:
+        '''
+        Find an item within a collection.
+
+        Returns the first item whose properties match all keyword arguments.
+
+        Returns None if no matching item is found.
+        '''
+
+        def all_equal(obj: object, props: dict) -> bool:
+            for key, value in props:
+                if getattr(obj, key, None) != value:
+                    return False
+
+            return True
+
+        props = kwargs.items()
+
+        for item in collection.values() if isinstance(collection, dict) else collection:
+            if all_equal(item, props):
+                return item
+
+        return None
+
+    def get(self, collection: Iterable, **kwargs) -> Any:
+        '''
+        Return an item from a collection.
+
+        Returns the first item whose properties match all keyword arguments.
+
+        Raises an Exception if no matching item is found.
+        '''
+
+        item = self.find(collection, **kwargs)
+
+        if item is None:
+            raise Exception('Failed to find item')
+
+        return item
+
+    def guild(self) -> MagicMock:
+        '''
+        Create a mock guild.
+        '''
+
+        guild = MagicMock(channels=[], text_channels=[], roles=[], _members={}, emojis=[])
+
+        # These are called by the discord.Member parameter converter.
+        guild.get_member_named.return_value = None
+        guild.query_members = AsyncMock(return_value=[])
+
+        guild.get_member = lambda discord_id: self.find(guild._members, id=discord_id)
+
+        return guild
+
+    def record(self, spec=Record, **kwargs) -> Any:
+        '''
+        Create a mock of a class that derives from Record.
+        '''
+
+        record = create_autospec(spec)
+
+        self.set_props(record, kwargs)
+
+        return record
+
+    def drone(self, drone_id: str | int, **kwargs) -> MagicMock:
+        '''
+        Create a mock drone record.
+        '''
+
+        if len(str(drone_id)) != 4:
+            raise RuntimeError('Drone IDs should be 4 digits.')
```

Comment:
This will lead to errors, if on enters a drone_id as an int like 0042, as the leading zeroes will be dropped. Is it maybe easier to just drop support for ints here, so we don't have to deal with those cases?

Otherwise I would kinda expect this function to perform padding with zeroes.